doc/manual/manual.tex

```diff
@@ -5023,6 +5023,34 @@ \subsubsection{Prescribed internal velocity constraints}
     \label{fig:prescribed-velocity}
 \end{figure}
 
+\subsubsection{Artificial viscosity smoothing}
+\label{sec:artificial-viscosity-smoothing}
+\textit{This section was contributed by Ryan Grove}\\
+
+Standard finite element discretizations of advection-diffusion equations introduce unphysical oscillations around steep gradients. Therefore, stabilization must be added to the discrete formulation to obtain correct solutions. In ASPECT, we use the Entropy Viscosity scheme developed by Guermond et al. in the paper \cite{guer11}. In this scheme, an artificial viscosity is calculated on every cell in and used to try to combat these oscillations that cause unwanted overshoot and undershoot.  More information about how ASPECT does this is located at \url{https://dealii.org/developer/doxygen/deal.II/step_31.html}.  \\
+
+Instead of just looking at an individual cell's artificial viscosity, improvements in the minimizing of the oscillations can be made by smoothing.  The motivation behind smoothing came from the idea of not only using a particular cell's artificial viscosity, but the artificial viscosity of neighboring cells as well.  The smoothing process refers to the comparing of the artificial viscosity of each cell with the artificial viscosity of each of the particular cell's neighboring cells and the maximum of this comparison is then used for future calculations.\\
+
+This feature can be turned on by setting the $Use \text{  } artificial \text{  }  viscosity$ flag inside the $Stabilization \text{  } parameters$ subsection inside the $Discretization$ subsection in your parameter file.
+Starting with the simple convection in a quarter of a 2d annulus cookbook in Section \ref{sec:shell-simple-2d}, a radial compositional field was added to help show the advantages of using the artificial viscosity smoothing feature.\\
+  
+By applying the following changes shown below to the parameters of the already existing file \begin{verbatim}cookbooks/shell_simple_2d/shell_simple_2d.prm, \end{verbatim} 
+it is possible to produce pictures of the simple convection in a quarter of a 2d annulus such as the ones visualized in
+(Figure \ref{fig:smoothing}).
+\lstinputlisting[language=prmfile]{cookbooks/shell_simple_2d_smoothing/shell_simple_2d_smoothing_piece.prm.out}
+\begin{figure}
+    \centering
+  \subfigure[]{
+    \includegraphics[width=.48\textwidth]{cookbooks/shell_simple_2d_smoothing/with_smoothing.png}}
+  ~
+  \subfigure[]{
+    \includegraphics[width=.48\textwidth]{cookbooks/shell_simple_2d_smoothing/without_smoothing.png}}
+    \caption{Example of the output of two similar runs.  The run on the left has the artificial viscosity smoothing turned on and the run on the right does not, as described in Section \ref{sec:artificial-viscosity-smoothing}.}
```

Comment:
Prefix the caption with an explanation which section of the text the figure goes with (because figures may float to the next page, or the one after that). E.g., say `Artificial viscosity smoothing: Example...`